src/scripts/download-images.js

```diff
@@ -0,0 +1,91 @@
+const fsExtra = require('fs-extra');
+const path = require('path');
+const { read } = require('to-vfile')
+const remark = require('remark');
+const mdx = require('remark-mdx');
+const visit = require('unist-util-visit')
+const https = require('https');
+const http = require('http');
+var mime = require('mime-types');
+const slugify = require('slugify');
+const glob = require('glob');
+
+
+const articlesPath = './src/pages/blog';
+// const articlesPath = './src/blog_vault';
+
+const ACCEPTED_FILES = ['jpg', 'jpeg', 'png', 'svg'];
+
+glob(articlesPath + '/*/*.mdx', {}, (err, files) => {
+    files.forEach( async (file) => {
+
+        const parentDirectory = path.dirname(file);
+
+        const mdxFile = await read(file)
+        //I process the file to get the AST
+        await remark()
+            .use(mdx)
+            .use(() => tree => {
+                 //I visit the AST to find images, searching for remote images
+                visit(tree, ['image'], async (node) => {
+                    // I check if it starts with either http or https
+                    if (node.url && node.url.startsWith('http')) {
+                        const title = node.title || node.alt || node.url.slice(node.url.length - 10);
+                        const slugTitle = slugify(title, {remove: /[*+~.()'"!?:@,]/g});
+
+                        const imagesPath = path.join(parentDirectory, 'img');
+                        const destFileWithoutExtension = path.join(imagesPath, slugTitle);
+
+                        try {
+                            await fsExtra.ensureDir(imagesPath);
+                            downloadFile(node.url, destFileWithoutExtension)
+                                .then(fileInfo => console.log(`Image ${slugTitle}.${fileInfo.extension} downloaded in ${imagesPath}`))
+                                .catch(err => console.error(`Error downloading image in ${imagesPath} from ${node.url}: `, err));
+
+                          } catch (e) {
+                            console.error(e)
+                          }
+
+
+                    }
+                })
+            })
+            .process(mdxFile)
+    });
+})
+
+
+function downloadFile (url, filePath) {
+    const proto = !url.charAt(4).localeCompare('s') ? https : http;
+  
+    return new Promise(function(resolve, reject) {
+        try {
+            
+            proto.get(url, function (response) {
```

Comment:
No easier to use libraries? The default http/https API always felt pretty cumbersome to me

Reply:
What about node-fetch? I used that now, it also follows redirects